integration-tests/tests/src/tests/sync/sync-session.ts

```diff
@@ -681,50 +662,43 @@ describe("SessionTest", () => {
       const credentials = Realm.Credentials.anonymous();
       const partition = generatePartition();
 
-      return this.app.logIn(credentials).then((user) => {
-        // Check valid input
-        const config1 = getSyncConfiguration(user, partition);
-        //@ts-expect-error internal field
-        config1.sync._sessionStopPolicy = "after-upload";
-
-        new Realm(config1).close();
-
-        const config2 = config1;
-        //@ts-expect-error internal field
-        config2.sync._sessionStopPolicy = "immediately";
-        new Realm(config2).close();
-
-        const config3 = config1;
-        //@ts-expect-error internal field
-        config3.sync._sessionStopPolicy = "never";
-        new Realm(config3).close();
-
-        // Invalid input
-        const config4 = config1;
-        //@ts-expect-error internal field
-        config4.sync._sessionStopPolicy = "foo";
-        expect(() => new Realm(config4)).throws();
-      });
+      const user = await this.app.logIn(credentials);
+      const config = getSyncConfiguration(user, partition);
+
+      // Check valid input
+
+      //@ts-expect-error internal field
+      config.sync._sessionStopPolicy = "after-upload";
+      new Realm(config).close();
+
+      //@ts-expect-error internal field
+      config.sync._sessionStopPolicy = "immediately";
+      new Realm(config).close();
+
+      //@ts-expect-error internal field
+      config.sync._sessionStopPolicy = "never";
+      new Realm(config).close();
+
+      //@ts-expect-error internal field
+      config.sync._sessionStopPolicy = "foo";
+      expect(() => new Realm(config)).throws();
     });
 
-    it("stop policy immediately succesfully removes session when realm is out of scope", async function (this: AppContext) {
+    it("stop policy immediately successfully removes session when realm is out of scope", async function (this: AppContext) {
```

Comment:
I'd assume `successfully` is the expectation in this sentence unless stated otherwise. What do you think?
```suggestion
    it("stop policy immediately removes session when realm is out of scope", async function (this: AppContext) {
```